src/librustc_mir/interpret/eval_context.rs

```diff
@@ -1716,49 +1716,53 @@ impl<'a, 'tcx, M: Machine<'tcx>> EvalContext<'a, 'tcx, M> {
-            for symbol in frame.symbols() {
-                if let Some(name) = symbol.name() {
-                    // unmangle the symbol via `to_string`
-                    let name = name.to_string();
-                    if name.starts_with("miri::after_analysis") {
-                        // don't report initialization gibberish
-                        break 'frames;
-                    } else if name.starts_with("backtrace::capture::Backtrace::new")
+    pub fn report(&self, e: &mut EvalError) {
+        if let Some(ref mut backtrace) = e.backtrace {
+            let mut trace_text = "\n\nAn error occurred in miri:\n".to_string();
+            let mut skip_init = true;
+            backtrace.resolve();
+            'frames: for (i, frame) in backtrace.frames().iter().enumerate() {
+                for symbol in frame.symbols() {
+                    if let Some(name) = symbol.name() {
+                        // unmangle the symbol via `to_string`
+                        let name = name.to_string();
+                        if name.starts_with("miri::after_analysis") {
+                            // don't report initialization gibberish
+                            break 'frames;
+                        } else if name.starts_with("backtrace::capture::Backtrace::new")
                             // debug mode produces funky symbol names
-                           || name.starts_with("backtrace::capture::{{impl}}::new") {
-                        // don't report backtrace internals
-                        skip_init = false;
-                        continue 'frames;
+                            || name.starts_with("backtrace::capture::{{impl}}::new") {
+                            // don't report backtrace internals
+                            skip_init = false;
+                            continue 'frames;
+                        }
                     }
                 }
-            }
-            if skip_init {
-                continue;
-            }
-            write!(trace_text, "{}\n", i).unwrap();
-            for symbol in frame.symbols() {
-                if let Some(name) = symbol.name() {
-                    write!(trace_text, "# {}\n", name).unwrap();
-                } else {
-                    write!(trace_text, "# <unknown>\n").unwrap();
-                }
-                if let Some(file_path) = symbol.filename() {
-                    write!(trace_text, "{}", file_path.display()).unwrap();
-                } else {
-                    write!(trace_text, "<unknown_file>").unwrap();
+                if skip_init {
+                    continue;
                 }
-                if let Some(line) = symbol.lineno() {
-                    write!(trace_text, ":{}\n", line).unwrap();
-                } else {
-                    write!(trace_text, "\n").unwrap();
+                for symbol in frame.symbols() {
+                    write!(trace_text, "{}: " , i).unwrap();
```

Comment:
the stack frame number is now reported for every symbol, instead of once for all symbols at that frame. Is that an intendend change?